plugins/parodos/src/components/Form/widgets/SelectWidget.tsx

```diff
@@ -1,7 +1,68 @@
-import React from 'react';
+import React, { useCallback, useEffect } from 'react';
 import { WidgetProps } from '@rjsf/utils';
 import { widgets } from './widgets';
+import { useUpdateSchema } from '../../../hooks/useUpdateSchema';
+import { assert } from 'assert-ts';
+import { errorApiRef, useApi } from '@backstage/core-plugin-api';
 
-export function SelectWidget(props: WidgetProps) {
-  return <widgets.SelectWidget {...props} variant="outlined" />;
+type ChangeEventHandlerArgs = Parameters<WidgetProps['onChange']>[0];
+
+export function SelectWidget({
+  disabled,
+  onChange,
+  formContext,
+  ...props
+}: WidgetProps) {
+  const errorApi = useApi(errorApiRef);
+  const valueProviderName = props.schema.valueProviderName;
+  const workflowDefinitionName = Object.keys(
+    props.registry.rootSchema.properties ?? {},
+  )[0];
+  const [{ loading, error }, updateSchema] = useUpdateSchema({
+    valueProviderName,
+    workflowDefinitionName,
+    updateSchema: formContext.updateSchema,
+  });
+
+  const changeHandler = useCallback(
+    async (value: ChangeEventHandlerArgs) => {
+      if (valueProviderName) {
+        assert(!!props.schema.title);
+
+        await updateSchema([
+          {
+            key: props.schema.title,
+            value,
+            workName: workflowDefinitionName,
+          },
+        ]);
+      }
+
+      onChange(value);
```

Comment:
Possible race-condition

Reply:
maybe we can disable all the fields in the form and enable them again after the onChange() call?

Reply:
The select is disabled by by the loading flag, see below.  Why would we need to disable the whole form

Reply:
sorry I thought the race-condition is between multiple fields in the same form.  @wKich  please help explain a bit more, Thank you